### PR TITLE
Deprecate mitchellh/go-homedir package in favor of os/user

### PR DESCRIPTION
**Pull request by arizvisa**
As per the conversation on #7036, this replaces all instances of mitchellh/go-homedir with an implementation based on os/user and removes the mitchellh/go-homedir package.

### CODE REVIEW

**Comment by arizvisa** [comment]
okay, fixed those dumb issues related to the version skew.

File: packer/config_file.go [hunk 1]
```diff
@@ -38,3 +40,46 @@ func ConfigTmpDir() (string, error) {
 	}
 	return td, nil
 }
+
+// Given a path, check to see if it's using ~ to reference a user directory.
+// If so, then replace that component with the requrested user's directory.
+func ExpandUser(path string) (string, error) {
```
Comment (azr) [inline packer/config_file.go]: Nice, I think this one deserves a test or two; will start testing this soon.

**Comment by arizvisa** [comment]
okay, added tests and rebased too. Hopefully the tests work out...

**Comment by arizvisa** [comment]
kk. committed your fix.

sure. whichever works. thanks!

**Comment by azr** [comment]
Here we go; thanks again @arizvisa 🙂

**Pull request merged**